### PR TITLE
Sprints/trois

### PR DESCRIPTION
**Pull request by pierluc-ssoucy**

### CODE REVIEW

File: src/Compliance.Solution/Solution/Entities/Task/FormXml/main/{04e44965-fddf-eb11-bacb-000d3af4a60d}.xml [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                        </labels>
+                      </cell>
+                    </row>
+                  </rows>
+                </section>
+              </sections>
+            </column>
+            <column width="33%">
+              <sections>
+                <section name="tab_2_section_2" showlabel="false" showbar="false" locklevel="0" id="{7a19c1aa-7ac9-5449-dcb0-208db3ecd341}" IsUserDefined="0" layout="varwidth" columns="11" labelwidth="115" labelid="{d8b3c781-ff03-4632-a7b5-2803f8f11936}">
```
Comment (EvoPheonix) [inline src/Compliance.Solution/Solution/Entities/Task/FormXml/main/{04e44965-fddf-eb11-bacb-000d3af4a60d}.xml:122]: Should this have a name?

File: src/Compliance.Solution/Solution/Other/ConnectionRoles.xml [hunk 2]
```diff
@@ -32,8 +32,8 @@
   </ConnectionRoles>
   <ConnectionRoleAssociations>
     <ConnectionRoleAssociation>
-      <connectionroleid>{6c2fe97d-27cd-eb11-bacc-0022486d7f1b}</connectionroleid>
-      <associatedconnectionroleid>{35cce071-27cd-eb11-bacc-0022486d7f1b}</associatedconnectionroleid>
+      <connectionroleid>{35cce071-27cd-eb11-bacc-0022486d7f1b}</connectionroleid>
+      <associatedconnectionroleid>{6c2fe97d-27cd-eb11-bacc-0022486d7f1b}</associatedconnectionroleid>
```
Comment (pierluc-ssoucy) [inline src/Compliance.Solution/Solution/Other/ConnectionRoles.xml:36]: is this something we need to worry about?

File: src/Compliance.Solution/Solution/Entities/opc_individualcomplaint_bp/FormXml/main/{ebfc2170-f3fe-47d9-b3d1-66475d7a96a6}.xml [hunk 3]
```diff
@@ -518,14 +526,14 @@
             </column>
           </columns>
         </tab>
-        <tab verticallayout="true" id="{7265e52b-2620-4736-ad8e-8376219e8abd}" name="StageStep105" locklevel="0" expanded="true" visible="false" labelid="{1dad1b6b-b4b5-4fb9-b815-4308ea090ce7}">
+        <tab verticallayout="true" id="{7265e52b-2620-4736-ad8e-8376219e8abd}" name="StageStep107" locklevel="0" expanded="true" visible="false" labelid="{1dad1b6b-b4b5-4fb9-b815-4308ea090ce7}">
```
Comment (EvoPheonix) [inline src/Compliance.Solution/Solution/Entities/opc_individualcomplaint_bp/FormXml/main/{ebfc2170-f3fe-47d9-b3d1-66475d7a96a6}.xml:529]: Should the tabs and sections here be named?

File: src/Compliance.Solution/Solution/SdkMessageProcessingSteps/{e7c47fba-4c12-eb11-a813-000d3a0c26f9}.xml [hunk 4]
```diff
@@ -1,18 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<SdkMessageProcessingStep Name="Compliance.Plugins.TimelineTransformationsPlugin: RetrieveTimelineWallRecords of  any Entity (Profiled)" SdkMessageProcessingStepId="{e7c47fba-4c12-eb11-a813-000d3a0c26f9}" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
```
Comment (pierluc-ssoucy) [inline src/Compliance.Solution/Solution/SdkMessageProcessingSteps/{e7c47fba-4c12-eb11-a813-000d3a0c26f9}.xml:2]: @GabrielFG This was fixed, but TimelineEvent plugin does not have a step anymore.

File: src/Compliance.Solution/Solution/Other/ConnectionRoles.xml (hunk 2 above)
Comment (nlelicanin) [reply src/Compliance.Solution/Solution/Other/ConnectionRoles.xml:36]: 35cce071... is the "Split From"
6c2fe97d... is the "Split To"

We created that "association" in the last sprint - so not sure why the ids "flipped"!? It seems to be running ok, though!

File: src/Compliance.WebResources/src/forms/IssueMainForm.test.ts [hunk 5]
```diff
@@ -139,6 +139,45 @@ describe("Issue - Main", () => {
...
+            // Act
+            await form.initializeComponents(mockContext);
+
+            // Assert
+            contextSpy.getFormContext().getControl("opc_accessrequestnumber").getVisible().should.equal(true);
+            contextSpy.getFormContext().getControl("opc_contact").getVisible().should.equal(true);
+            contextSpy.getFormContext().getControl("subgrid_accessrequestdocuments").getVisible().should.equal(false);
+        });
+
+        it("is changed to 'TimeLimitCorrection', it should ONLY SHOW releveant fields", async () => {
```
Comment (EvoPheonix) [inline src/Compliance.WebResources/src/forms/IssueMainForm.test.ts:155]: @pierluc-ssoucy please make the description more explicit for the relevant fields

File: src/Compliance.WebResources/src/forms/IssueMainForm.test.ts [hunk 6]
```diff
@@ -217,4 +256,64 @@ describe("Issue - Main", () => {
...
+            formContext.getAttribute("opc_disposition").setValue([opc_issuedisposition.Wellfounded]);
+
+            // Act
+            await form.initializeComponents(mockContext);
+            formContext.getAttribute("opc_disposition").fireOnChange();
+
+            // Assert
+            chai.assert.isTrue(contextSpy.getFormContext().getAttribute("opc_dispositiondate").getValue() != null);
+        });
+    });
```
Comment (pierluc-ssoucy) [inline src/Compliance.WebResources/src/forms/IssueMainForm.test.ts:318]: @nlelicanin we are already in a spy so we could use the preferred "should" notation.

File: src/Compliance.Solution/Solution/Entities/Task/FormXml/main/{04e44965-fddf-eb11-bacb-000d3af4a60d}.xml (hunk 1 above)
Comment (pierluc-ssoucy) [reply src/Compliance.Solution/Solution/Entities/Task/FormXml/main/{04e44965-fddf-eb11-bacb-000d3af4a60d}.xml:122]: ideally yes but not critical if not we don't have TS references

File: src/Compliance.WebResources/src/forms/IssueMainForm.test.ts (hunk 6 above)
Comment (nlelicanin) [reply src/Compliance.WebResources/src/forms/IssueMainForm.test.ts:318]: As discussed, ".should.be.null" errors out, so instead it's ok to use chai asserts.

File: src/Compliance.Solution/Solution/Entities/Task/FormXml/main/{04e44965-fddf-eb11-bacb-000d3af4a60d}.xml [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+              </labels>
+            </cell>
+          </row>
+        </rows>
+      </footer>
+    </form>
+    <IsCustomizable>1</IsCustomizable>
+    <CanBeDeleted>1</CanBeDeleted>
+    <LocalizedNames>
+      <LocalizedName description=" main form" languagecode="1033" />
```
Comment (JonathanThomassin) [inline src/Compliance.Solution/Solution/Entities/Task/FormXml/main/{04e44965-fddf-eb11-bacb-000d3af4a60d}.xml:226]: Contact and Team also have copies of their main form. We should probably rename this with the same format as the others. Also there is a space at the beginning.
**ComplianceTask**
ComplianceContact
ComplianceTeam

File: src/Compliance.Solution/Solution/Entities/Task/FormXml/quick/{1a06871c-03e4-eb11-bacb-0022486d87ef}.xml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        <Role Id="{119f245c-3cc8-4b62-b31c-d1a046ced15d}" />
+        <Role Id="{b3b64455-85a8-439d-9716-66fe9c912264}" />
+        <Role Id="{33f05dff-1a50-42f0-8902-911b74bd2025}" />
+        <Role Id="{c3a77b7c-0478-4c37-ba86-c2dadc6c2dcf}" />
+      </DisplayConditions>
+    </form>
+    <IsCustomizable>1</IsCustomizable>
+    <CanBeDeleted>1</CanBeDeleted>
+    <LocalizedNames>
+      <LocalizedName description="quick view form" languagecode="1033" />
```
Comment (JonathanThomassin) [inline src/Compliance.Solution/Solution/Entities/Task/FormXml/quick/{1a06871c-03e4-eb11-bacb-0022486d87ef}.xml:49]: We should probably rename this to be more like the contact one.

File: src/Compliance.Solution/Solution/Entities/opc_complaint/SavedQueries/{026798e0-d3a5-eb11-9442-000d3af48827}.xml [hunk 9]
```diff
@@ -43,6 +44,9 @@
             <condition attribute="ownerid" operator="eq-userid" />
           </filter>
           <attribute name="opc_lastactionid" />
+          <link-entity alias="a_22320e233b804a70b2a1b66e5802c604" name="task" to="opc_lastactionid" from="activityid" link-type="outer" visible="false">
```
Comment (JonathanThomassin) [inline src/Compliance.Solution/Solution/Entities/opc_complaint/SavedQueries/{026798e0-d3a5-eb11-9442-000d3af48827}.xml:47]: Should we rename the alias?

**Pull request merged**